core/benches/bench.rs

```diff
@@ -0,0 +1,216 @@
+        b.iter(|| put_cached(i))
+    });
+    group.bench_with_input(BenchmarkId::new("Deref", 0), &test_values, |b, i| {
+        b.iter(|| deref_cached(i))
+    });
+    group.finish();
+}
+
+fn bench_clear_cached(c: &mut Criterion) {
+    let mut group = c.benchmark_group("ClearMustOutperformPopInCachedCase");
+    let test_values = [b'A', b'B', b'C', b'D', b'E', b'F'];
+    group.bench_with_input(BenchmarkId::new("Clear", 0), &test_values, |b, i| {
+        b.iter(|| clear(i))
+    });
+    group.bench_with_input(BenchmarkId::new("PopAll", 0), &test_values, |b, i| {
+        b.iter(|| pop_all(i))
+    });
+    group.finish();
+}
+
+/// The manual drop is used to prevent the `vec` from being written back to storage.
+/// This is so that it can be reused in the next benchmark instance, without the
+/// storage flush overhead.
+fn manual_drop(vec: StorageVec<u8>) {
+    ManuallyDrop::new(vec);
+}
+
+/// In this case we lazily load the vec from storage using `pull_spread`.
+/// This will just load lazily and won't pull anything from the storage.
+fn clear_lazy() {
+    let root_key = Key::from([0x00; 32]);
+    let mut vec =
+        <StorageVec<u8> as SpreadLayout>::pull_spread(&mut KeyPtr::from(root_key));
+    black_box(vec.clear());
+    assert!(vec.is_empty());
+    manual_drop(vec);
+}
+
+/// In this case we lazily load the vec from storage using `pull_spread`.
+/// This will just load lazily and won't pull anything from the storage.
+/// `pop` will then result in loading from storage.
+fn pop_all_lazy() {
+    let root_key = Key::from([0x00; 32]);
+    let mut vec =
+        <StorageVec<u8> as SpreadLayout>::pull_spread(&mut KeyPtr::from(root_key));
+    while let Some(_) = black_box(vec.pop()) {}
+    assert!(vec.is_empty());
+    manual_drop(vec);
+}
+
+/// In this case we lazily load the vec from storage using `pull_spread`.
+/// This will just load lazily and won't pull anything from the storage.
+/// The `deref` will then load from storage.
+fn deref_lazy() {
+    let root_key = Key::from([0x00; 32]);
+    let mut vec =
+        <StorageVec<u8> as SpreadLayout>::pull_spread(&mut KeyPtr::from(root_key));
+    for index in 0..vec.len() {
+        *vec.get_mut(index).unwrap() = b'X';
+    }
+    assert_eq_slice(&vec, &[b'X', b'X', b'X', b'X', b'X', b'X']);
+    manual_drop(vec);
+}
+
+/// In this case we lazily load the vec from storage using `pull_spread`.
+/// This will just load lazily and won't pull anything from the storage.
+/// The `vec.set()` will not load anything from storage.
+fn put_lazy() {
+    let root_key = Key::from([0x00; 32]);
+    let mut vec =
+        <StorageVec<u8> as SpreadLayout>::pull_spread(&mut KeyPtr::from(root_key));
+    for index in 0..vec.len() {
+        vec.set(index, b'X');
+    }
+    assert_eq_slice(&vec, &[b'X', b'X', b'X', b'X', b'X', b'X']);
+    manual_drop(vec);
+}
+
+/// Create a vec and push it to storage.
+fn create_and_store() {
+    let test_values = [b'A', b'B', b'C', b'D', b'E', b'F'];
```

Comment:
Try increasing the `test_values` to some more elements.
Or: Create multiple sizes, small, medium and large to benchmark for different sizes.
Maybe small = 3 elements, medium = 30 elements, large = 300 elements.